wordpress-mobile/WordPress-Android
# Issue/13446 clean up comment related analytics

Fixes #13446

This PR updates comment related analytics.

Since we are probably going to do a global rework of comment logic (which will also include analytics), I decided to not overthink it and integrated most of the tracking with existing AnalyticsUtils code.

I kept the existing events for now and fixed the issue with site comment events being bundled with notification ones.

What changed:
- We differentiating the source of comment action (eg. Notifications, Reader, Site Comments)
- Simplified names.
- Adding site type when tracking with Reader's Post (eg. p2 or blog).
- Added a couple of extra events (eg, Reply, Delete, View)

To test:
- Confirm the new events in `AnalyticsTracker`.
Place breakpoints in the following `AnalyticsUtils` methods:
- trackCommentActionWithSiteDetails
- trackCommentActionWithReaderPostDetails
- trackCommentReplyWithDetails (this will work for non-reader comment replies)
And confirm that they are called with correct parameters and at the correct timing.

PR submission checklist:
